### DISCUSSION TITLE
Return void from GraphQL Mutation

### DISCUSSION THREAD

**Discussion by zbhavyai** [quarkusio/quarkus#25381] [Q&A] [answered]
Hello,

I am testing SmallRye GraphQL Mutations with Quarkus. I have a scenario where method in a service class returns void, and I would like to have similar setup with the GraphQL wrapper, where a mutation returns nothing after an operation is performed.

However, I am getting `java.lang.RuntimeException: Failed to start quarkus at io.quarkus.runner.ApplicationImpl.<clinit>(Unknown Source) graphql.AssertException: type Void not found in schema` when I try to compile the code.

I made a small reproducer to replicate the exception I am getting. It is available at [https://github.com/zbhavyai/quarkus-graphql-void-reproducer](https://github.com/zbhavyai/quarkus-graphql-void-reproducer).

I am using Quarkus versions 2.8.2.Final.

So, I would like to know, if it is a GraphQL limitation, or am I missing something? Because return void from a REST endpoint works absolutely fine.

Thank you.

[a comment by an automated account is not shown]

**Reply by phillip-kruger** [reply]
By default GraphQL does not support returning nothing. That being said, we should be able to add support for the `Void` object. You are welcome to open an issue for it in the SmallRye project.

**Reply by zbhavyai** [reply]
Thank you so much @phillip-kruger. I opened an issue here - [https://github.com/smallrye/smallrye-graphql/issues/1383](https://github.com/smallrye/smallrye-graphql/issues/1383).
